src/pages/Import.js

```diff
@@ -0,0 +1,186 @@
+          borderRadius: 5, backgroundColor: 'white', padding: '1rem',
+          boxShadow: `0 0 2px 0 rgba(0,0,0,0.12), ${Style.standardShadow}`
+        }
+      },
+      [
+        div({ style: { fontSize: 16 } }, `From Dockstore - ${path}`),
+        div({}, `V. ${version}`),
+        div({
+            style: {
+              display: 'flex', alignItems: 'center',
+              margin: '1rem 0', color: Style.colors.warning
+            }
+          },
+          [
+            icon('warning-standard', { class: 'is-solid', size: 32, style: { marginRight: '0.5rem', flex: '0 0 auto' } }),
+            mutabilityWarning
+          ]),
+        h(Collapse, { title: 'REVIEW WDL' },
+          [h(WDLViewer, { wdl })])
+      ]
+    )
+  }
+
+  renderWorkspaceArea = () => {
+    const { selectedWorkspace, workspaces, importError } = this.state
+
+    return div({},
+      [
+        h(Select, {
+          clearable: false,
+          searchable: true,
+          disabled: !workspaces,
+          placeholder: workspaces ? 'Select a workspace' : 'Loading workspaces...',
+          value: selectedWorkspace,
+          onChange: selectedWorkspace => this.setState({ selectedWorkspace }),
+          options: _.map(workspaces, workspace => {
+            const { name } = workspace.workspace
+            return { value: name, label: name, data: workspace }
+          })
+        }),
+        buttonPrimary(
+          {
+            style: { marginTop: '1rem' },
+            disabled: !selectedWorkspace,
+            onClick: () => this.import()
+          },
+          'Import'),
+        importError ?
+          div({
+            style: { marginTop: '1rem', color: Style.colors.error }
+          }, [
+            icon('error'),
+            JSON.parse(importError).message
+          ]) : null
+      ]
+    )
+  }
+
+  import = () => {
+    const { selectedWorkspace: { data: { workspace: { namespace, name } } } } = this.state
+    const { path, version } = this.props
+
+    Rawls.workspace(namespace, name).methodConfigs().importFromDocker({
+      namespace, name: _.last(path.split('/')), rootEntityType: 'participant',
+      // the line of shame:
+      inputs: {}, outputs: {}, prerequisites: {}, methodConfigVersion: 1, deleted: false,
+      methodRepoMethod: {
+        sourceRepo: 'dockstore',
+        methodPath: path,
+        methodVersion: version
+      }
+    }).then(
+      () => Nav.goToPath('workspace', namespace, name, 'tools'),
+      importError => this.setState({ importError })
+    )
+  }
+
+  renderError = () => {
+    const { loadError } = this.state
+
+    return h(Fragment, [
+      div(wdlLoadError),
+      div(`Error: ${loadError}`)
+    ])
+  }
+}
+
+
+class Importer extends Component {
+  render() {
+    const { source } = this.props
+
+    return h(Fragment, [
+      h(TopBar, { title: 'Import' }),
+      Utils.cond(
+        [source === 'dockstore', this.renderDockstore],
+        () => `Unknown source '${source}'`
```

Comment:
What is this indirection for? Do we know for sure that we will have multiple import workflows on this page?